html/js/tree/ISet.js

```diff
@@ -0,0 +1,151 @@
+GTE.TREE = (function (parentModule) {
+    "use strict";
+
+    /**
+    * Creates a new ISet.
+    * @class
+    * @param {ISet} [parent] Parent ISet. If null, this is root.
+    * @param  {Number} [numOfNodes] Number of nodes of this new information set. If null 1
+    */
+    function ISet(parent, numOfNodes) {
+        this.parent = parent;
+        if (this.parent !== null) this.parent.addChild(this);
+        this.numOfNodes = numOfNodes || 1;
+         // TODO #11
+        var newNodeLevel;
+        if (parent === null) {
+            newNodeLevel = 0;
+        } else {
+            newNodeLevel = parent.node.level + 1;
+        }
+        this.node = new GTE.TREE.Node(this, newNodeLevel);
+        this.moves = [];
+    }
+
+    // TODO #12
+    /**
+    * ToString function
+    */
+    ISet.prototype.toString = function () {
+        return "ISet: " + "parent " + this.parent + "; numOfNodes: " + this.numOfNodes + "; moves: " + this.moves;
+    };
+
+    /**
+    * Function that adds a node to ISet
+    * @param {Node} node Node to add
+    */
+    ISet.prototype.addNode = function (node) {
+        this.nodes.push(node);
+    };
+
+    /**
+    * Function that adds a child ISet to the ISet
+    * @param {ISet} childISet ISet to add
+    */
+    ISet.prototype.addChild = function (childISet) {
+        // Create a move that leads to the child ISet
+        this.moves.push(new GTE.TREE.Move(this, childISet));
+    };
+
+    /**
+    * Function that finds the move that leads to a child ISet
+    * @param {ISet}   childISet ISet to look for
+    * @return {Number}          The index of the move in the list of
+    *                           moves that contains the given iSet 
+    */
+    ISet.prototype.findMoveThatLeadsTo = function (iSet) {
+        for (var indexInList = 0; indexInList < this.moves.length; indexInList++) {
+            if (this.moves[indexInList].child == iSet) return indexInList;
+        };
+        return -1;
+    };
+
+    /**
+    * Function that removes the move that contains a children iSet
+    * @param {ISet} iSetToDelete Child ISet to remove
+    */
+    ISet.prototype.removeChild = function (iSetToDelete) {
+        // Find the move that leads to this ISet
+        var indexInList = this.findMoveThatLeadsTo(iSetToDelete);
+        if (indexInList > -1) {
+            this.moves.splice(indexInList, 1);
+        }
```

Comment:
This will need to be changed once we figure out what to do with the previous comment.